### ISSUE TITLE
[Conditional mapping] Numbers are treated as strings when target is string

### ISSUE THREAD

**Issue by mmelko** [atlasmap/atlasmap#997]
Numbers are treated as strings when target is string. 
Result of expression: 
`integerA = 10
integerB = -10
 if(integerA < integerB, true, false)`
is `true` when target is string and `false` when it's number field. So it looks like conversion of fields is happening before expression evaluation. 
When you look into mapping preview everything works as expected.

**Comment by igarashitm** [comment]
So if target field type is `BOOLEAN` it works as expected. It rather sounds like about type conversion from `BOOLEAN` to `STRING` vs. `BOOLEAN` to `NUMBER`, which is expected to happen after expression evaluation. I think this is also not p0 as expecting `BOOLEAN` to `NUMBER` conversion is an edge case.

**Comment by igarashitm** [comment]
And at least `BOOLEAN` to `INTEGER` conversion after expression works as expected. Added a testcase https://github.com/atlasmap/atlasmap/pull/1000/files#diff-1737b36f8526b377af54d66b498d5560R102

**Comment by igarashitm** [comment]
Oops it was opposite, I'll try `BOOLEAN` to `STRING`

**Comment by igarashitm** [comment]
`BOOLEAN` to `STRING` case still works as expected. Here is a testcase https://github.com/atlasmap/atlasmap/pull/1001/files#diff-1737b36f8526b377af54d66b498d5560R102

**Comment by mmelko** [comment]
I personally don't agree that boolean to number is edge case.. it's basically only meaningful usecase for `<` `>` operators. Anyway .. what could be confusing for customers is that it's working properly in previews.

**Comment by igarashitm** [comment]
But what does "boolean to number" mean in real world? Anyway a testcase I just added shows even "boolean to string" after expression works as expected. It just converts into string literal "true" and "false". Can you check what's different from your case?

**Comment by mmelko** [comment]
Okay so I pulled latest master, run tests and everything is working fine. So it looks either it was mysteriously fixed or I had some typo there.

Just for the record this was my original expression: `if(@{sourceJsonInteger} < -10, @{sourceJsonString}, @{sourceJsonDouble} ) ` where `sourceJsonInteger=10` and it always returned `sourceJsonDouble`, but now it's working fine.  

Anyway sorry for confusion. Closing this for now.
